test/integration/testpaths/testpath_volume_snapshot.py

```diff
@@ -0,0 +1,745 @@
+            "mount -t ext3 %s1 %s" % (
+                self.testdata["volume_write_path"][
+                    new_virtual_machine.hypervisor][disk_type],
+                self.testdata["data_write_paths"]["mount_dir"]
+            ),
+            ]
+
+    for c in cmds:
+        self.debug("Command: %s" % c)
+        result = ssh.execute(c)
+        self.debug(result)
+
+    returned_data_0 = ssh.execute(
+        "cat %s/%s/%s/%s" % (
+            self.testdata["data_write_paths"]["mount_dir"],
+            self.testdata["data_write_paths"]["sub_dir"],
+            self.testdata["data_write_paths"]["sub_lvl_dir1"],
+            self.testdata["data_write_paths"]["random_data"]
+        ))
+
+    n = hashlib.md5()
+    n.update(returned_data_0[0])
+    ckecksum_returned_data_0 = n.hexdigest()
+
+    self.debug("returned_data_0: %s" % returned_data_0[0])
+
+    # Verify returned data
+    self.assertEqual(
+        original_checksum,
+        ckecksum_returned_data_0,
+        "Cheskum does not match with checksum of original data"
+    )
+
+    # Unmount the Sec Storage
+    cmds = [
+        "umount %s" % (self.testdata["data_write_paths"]["mount_dir"]),
+    ]
+
+    for c in cmds:
+        self.debug("Command: %s" % c)
+        ssh.execute(c)
+
+    if new_vm:
+        new_virtual_machine.detach_volume(
+            self.apiclient,
+            disk
+        )
+
+        new_virtual_machine.delete(self.apiclient)
+
+    return
+
+
+class TestVolumeSnapshot(cloudstackTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        testClient = super(TestVolumeSnapshot, cls).getClsTestClient()
+        cls.apiclient = testClient.getApiClient()
+        cls.testdata = testClient.getParsedTestDataConfig()
+        cls.hypervisor = cls.testClient.getHypervisorInfo()
+
+        # Get Zone, Domain and templates
+        cls.domain = get_domain(cls.apiclient)
+        cls.zone = get_zone(cls.apiclient, testClient.getZoneForTests())
+
+        cls.template = get_template(
+            cls.apiclient,
+            cls.zone.id,
+            cls.testdata["ostype"])
+
+        cls._cleanup = []
+
+        if cls.hypervisor.lower() not in [
+                "vmware",
+                "kvm",
+                "xenserver"]:
+            raise unittest.SkipTest(
+                "Storage migration not supported on %s" %
+                cls.hypervisor)
+
+        try:
+
+            # Create an account
+            cls.account = Account.create(
+                cls.apiclient,
+                cls.testdata["account"],
+                domainid=cls.domain.id
+            )
+            cls._cleanup.append(cls.account)
+
+            # Create user api client of the account
+            cls.userapiclient = testClient.getUserApiClient(
+                UserName=cls.account.name,
+                DomainName=cls.account.domain
+            )
+
+            # Create Service offering
+
+            cls.service_offering_cluster1 = ServiceOffering.create(
```

Comment:
Modify service offering name. Is this specific for CWPS? If not, change to default name.